### PR TITLE
Saas 13.5 fixes tde

### PR DESCRIPTION
**Pull request by tde-banana-odoo**
Task ID-2326281 (context keys use cleaning)
Task ID-2288458 (creation message)
Task ID-2327096 (code cleaning)
PR #56631
PR odoo/enterprise#12707

### CODE REVIEW

File: addons/mail/models/mail_thread.py [hunk 1]
```diff
@@ -835,10 +800,7 @@ def _routing_check_route(self, message, message_dict, route, raise_exception=Tru
                 obj = self.env[alias.alias_parent_model_id.model].browse(alias.alias_parent_thread_id)
             else:
                 obj = self.env[model]
-            if hasattr(obj, '_alias_check_contact'):
-                error_code = obj._alias_check_contact(message, message_dict, alias)
-            else:
-                error_code = self.env['mail.alias.mixin']._alias_check_contact_on_record(obj, message, message_dict, alias)
+            error_code = obj._alias_check_contact(message, message_dict, alias)
             if error_code is not True:
```
Comment (awa-odoo) [inline addons/mail/models/mail_thread.py:804]: I lack a bit of context but there seems to be a mismatch of naming here.
"_alias_check_contact", returning an "error_code" (wow), that is ignored if... True? (wow²).

Suggestion:
"_get_alias_access_error" returning an error string (as currently) or False if no error occured.
And you would check "if error_message" here.
Makes a bit more sense, don't you think?
(Still not a fan of my naming proposition but maybe you will have better).

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by seb-odoo** [comment]
Excellent to finally remove the "created" body.

This does so much more than just avoiding the "duplicate" text:
- subtype description is translatable whereas hardcoded body was not
- posting body was actually slow in batch record creation due to sanitize of html field
- cleaner db with tracking meta data not mixed with user content

File: addons/mail/models/mail_thread.py [hunk 2]
```diff
@@ -288,14 +284,16 @@ def create(self, vals_list):
...
-                    thread.sudo().message_post(body=body, subtype_id=subtype.id, author_id=self.env.user.partner_id.id)
+                    thread.sudo().message_post(subtype_id=subtype.id, author_id=self.env.user.partner_id.id)
                 else:
                     threads_no_subtype += thread
             if threads_no_subtype:
-                threads_no_subtype._message_log_batch(bodies={t.id: body for t in threads_no_subtype})
+                bodies = dict(
+                    (thread.id, thread._creation_message())
+                    for thread in threads_no_subtype)
+                threads_no_subtype._message_log_batch(bodies=bodies)
```
Comment (seb-odoo) [inline addons/mail/models/mail_thread.py:296]: I wish we could get rid of those body too but I guess that might be asking too much.

**Pull request closed without merging**